src/pages/CoinPage.js

```diff
@@ -11,31 +11,28 @@ import arrowLeft from '../images/arrow-left.svg';
 import arrowUp from '../images/arrow-up.svg';
 import arrowDown from '../images/arrow-down.svg';
 import spinner from '../images/spinner.svg';
-import useFetch from '../hooks/useFetch.js';
 
 function CoinPage({ coin, title, currency, toggleBookmark }) {
-  const [days, setDays] = useState(1);
+  const days = useStore(state => state.days);
+  const setDays = useStore(state => state.setDays);
+  const getData = useStore(state => state.getData);
+  const chartHistory = useStore(state => state.chartHistory);
 
-  const [
-    cryptoHistory,
-    isLoadingCryptoHistory,
-    errorCryptoHistory,
-    fetchCryptoHistory,
-  ] = useFetch(
-    `https://api.coingecko.com/api/v3/coins/${coin.id}/market_chart?vs_currency=${currency}&days=${days}`
-  );
-
-  const [coins, isLoadingCoins, errorCoins, fetchCoins] = useFetch(
-    `https://api.coingecko.com/api/v3/coins/markets?vs_currency=${currency}&order=market_cap_desc&per_page=100&page=1&sparkline=false`
-  );
+  useEffect(() => {
+    getData(
+      `https://api.coingecko.com/api/v3/coins/${coin.id}/market_chart?vs_currency=${currency}&days=${days}`,
+      'chartHistory'
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currency, days]);
 
-  if (isLoadingCoins && isLoadingCryptoHistory) {
+  if (coin.loading) {
     return <SpinnerLogo src={spinner} height="80" width="80"></SpinnerLogo>;
   }
```

Comment:
Same as above but as I said this is totally up to you as it is just another way to write that bit of code...

Reply:
Cheers for checking! I will leave it like that, so that it is really obvious to see that its just getting rendered in that case